nexus/db-queries/src/transaction_retry.rs

```diff
@@ -179,6 +185,48 @@ impl oximeter::Producer for Producer {
     }
 }
 
+/// Helper utility to have a similar interface to RetryHelper (also emitting
+/// probes) but for non-retryable transactions.
+pub struct NonRetryHelper {
+    log: Logger,
+    name: &'static str,
+}
+
+impl NonRetryHelper {
+    pub(crate) fn new(log: &Logger, name: &'static str) -> Self {
+        Self { log: log.new(o!("transaction" => name)), name }
+    }
+
+    /// Calls the function "f" in an asynchronous, non-retryable transaction.
+    pub async fn transaction<R, E, Func, Fut>(
+        self,
+        conn: &async_bb8_diesel::Connection<crate::db::DbConnection>,
+        f: Func,
+    ) -> Result<R, E>
+    where
+        R: Send + 'static,
+        E: From<DieselError> + std::fmt::Debug + Send + 'static,
+        Fut: std::future::Future<Output = Result<R, E>> + Send,
+        Func: FnOnce(async_bb8_diesel::Connection<crate::db::DbConnection>) -> Fut
+            + Send
+            + Sync,
+    {
+        crate::probes::transaction__start!(|| {
```

Comment:
Do we care about distinguishing retryable and non-retryable transactions in the probe?

Reply:
I'd be fine adding this as a follow-up, but I think the "name of the transaction" acts as a sufficient distinguisher for now?